tools/statvar_importer/eval_functions.py

```diff
@@ -97,20 +142,44 @@ def str_to_camel_case(input_string: str,
 def evaluate_statement(eval_str: str,
                        variables: dict = {},
                        functions: dict = EVAL_GLOBALS) -> (str, str):
-    """Returns the tuple: (variable, result) after evaluating statement in eval.
-
-   Args:
-     eval_str: string with statement to be evaluated of the form:
-       'variable=statement' if the variable is not specified, an empty string is
-       retured as variable.
-      variables: dictionary of variables and values to be used in statement.
-      functions: dictionary of global functoins that can be invoked within
-        statement.
-
-  Returns:
-      tuple of the (variable , result) after evaluating the statement.
-      in case of exception during eval, None is returned as result
-  """
+    """Evaluates a Python expression and returns the result.
+
+    This function is a safe wrapper around Python's `eval()` built-in. It
+    is designed to execute simple expressions, often used for data
+    transformations. The expression can optionally assign a value to a
+    variable.
+
+    Args:
+        eval_str: The string containing the expression to be evaluated, in the
+          format 'variable = statement' or just 'statement'.
+        variables: A dictionary of variables and their values that are
+          accessible within the `eval()` context.
+        functions: A dictionary of functions that are accessible within the
+          `eval()` context. Defaults to `EVAL_GLOBALS`.
+
+    Returns:
+        A tuple containing the variable name and the result of the evaluation.
+        - If `eval_str` includes a variable assignment, the tuple will be
+          (`variable_name`, `result`).
+        - If `eval_str` is a simple statement, the tuple will be
+          ('', `result`).
+        - If the evaluation raises a `SyntaxError`, `NameError`, `ValueError`,
+          or `TypeError`, the result will be `None`.
+
+    Examples:
+        >>> # Simple arithmetic
+        >>> evaluate_statement('num = 1 + x', {'x': 2})
+        ('num', 3)
+        >>> # Using a function from EVAL_GLOBALS
+        >>> evaluate_statement('date = format_date(data)', {'data': '2022, Jan 1st'})
+        ('date', '2022-01-01')
+        >>> # Statement without a variable
+        >>> evaluate_statement('2 * 2')
+        ('', 4)
+        >>> # Invalid expression
+        >>> evaluate_statement('name = 1 + "2"')
+        ('name', None)
+    """
     variable = ''
     statement = eval_str
     if '=' in eval_str:
```

Comment:
Can you extend this to catch any exception. It missed IndexError for some imports

Reply:
Done.